llvm/lib/Support/YAMLParser.cpp

```diff
@@ -1848,13 +1869,15 @@ bool Scanner::fetchMoreTokens() {
   if (*Current == ',')
     return scanFlowEntry();
 
-  if (*Current == '-' && isBlankOrBreak(Current + 1))
+  if (*Current == '-' && (isBlankOrBreak(Current + 1) || Current + 1 == End))
```

Comment:
Can you add a test for this change?

Reply:
Ah, it makes sense now why there wasn't a new test; thank you for adding an explicit test for the shape of the result, I think that's an improvement regardless